synapse/rest/client/v1/login.py

```diff
@@ -60,10 +60,18 @@ def login_id_thirdparty_from_phone(identifier):
 
     Returns: Login identifier dict of type 'm.id.threepid'
     """
-    if "country" not in identifier or "number" not in identifier:
+    if "country" not in identifier or (
+        # XXX: We used to require `number` instead of `phone`. The spec
```

Comment:
Why `XXX`? I usually reserve that for something that's an action, maybe something like:

> The specification requires a `phone` field, but Synapse used to require a `number` field. Accept both for backwards compatibility.

Reply:
I guess it was more intended as a `TODO`, we should remove this later.

...but, we're probably never going to remove it, so I think your suggestion does work better.